From ce9fe511ccb5d5463c2815853a7d690b116cba26 Mon Sep 17 00:00:00 2001
From: Shaofeng Tang <shaofeng.tang@intel.com>
Date: Wed, 25 Nov 2020 15:20:35 +0800
Subject: [PATCH] Skip CFI checking for 2 comparator

The function pointer "void * priv" are defined in <linux/list_sort.h>
we can not change it to any other type, althouth this pointer is not
used at all.

Tracked-On: OAM-94511 and OAM-94603
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>
---
 drivers/gpu/drm/drm_modes.c                 | 4 ++--
 drivers/gpu/drm/i915/gt/intel_engine_user.c | 6 +++---
 drivers/gpu/drm/i915/gt/intel_engine_user.h | 2 +-
 include/drm/drm_modes.h                     | 2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/drm_modes.c b/drivers/gpu/drm/drm_modes.c
index 3fd35e6b9d53..c5f44dcfe000 100644
--- a/drivers/gpu/drm/drm_modes.c
+++ b/drivers/gpu/drm/drm_modes.c
@@ -1319,7 +1319,7 @@ EXPORT_SYMBOL(drm_mode_prune_invalid);
  * Negative if @lh_a is better than @lh_b, zero if they're equivalent, or
  * positive if @lh_b is better than @lh_a.
  */
-static int drm_mode_compare(void *priv, struct list_head *lh_a, struct list_head *lh_b)
+static int __nocfi drm_mode_compare(void *priv, struct list_head *lh_a, struct list_head *lh_b)
 {
 	struct drm_display_mode *a = list_entry(lh_a, struct drm_display_mode, head);
 	struct drm_display_mode *b = list_entry(lh_b, struct drm_display_mode, head);
@@ -1347,7 +1347,7 @@ static int drm_mode_compare(void *priv, struct list_head *lh_a, struct list_head
  *
  * Sort @mode_list by favorability, moving good modes to the head of the list.
  */
-void drm_mode_sort(struct list_head *mode_list)
+void __nocfi drm_mode_sort(struct list_head *mode_list)
 {
 	list_sort(NULL, mode_list, drm_mode_compare);
 }
diff --git a/drivers/gpu/drm/i915/gt/intel_engine_user.c b/drivers/gpu/drm/i915/gt/intel_engine_user.c
index 848decee9066..d4bac9ce51ae 100644
--- a/drivers/gpu/drm/i915/gt/intel_engine_user.c
+++ b/drivers/gpu/drm/i915/gt/intel_engine_user.c
@@ -49,7 +49,7 @@ static const u8 uabi_classes[] = {
 	[VIDEO_ENHANCEMENT_CLASS] = I915_ENGINE_CLASS_VIDEO_ENHANCE,
 };
 
-static int engine_cmp(void *priv, struct list_head *A, struct list_head *B)
+static int __nocfi engine_cmp(void *priv, struct list_head *A, struct list_head *B)
 {
 	const struct intel_engine_cs *a =
 		container_of((struct rb_node *)A, typeof(*a), uabi_node);
@@ -74,7 +74,7 @@ static struct llist_node *get_engines(struct drm_i915_private *i915)
 	return llist_del_all((struct llist_head *)&i915->uabi_engines);
 }
 
-static void sort_engines(struct drm_i915_private *i915,
+static void __nocfi sort_engines(struct drm_i915_private *i915,
 			 struct list_head *engines)
 {
 	struct llist_node *pos, *next;
@@ -183,7 +183,7 @@ static void add_legacy_ring(struct legacy_ring *ring,
 		ring->instance++;
 }
 
-void intel_engines_driver_register(struct drm_i915_private *i915)
+void __nocfi intel_engines_driver_register(struct drm_i915_private *i915)
 {
 	struct legacy_ring ring = {};
 	u8 uabi_instances[4] = {};
diff --git a/drivers/gpu/drm/i915/gt/intel_engine_user.h b/drivers/gpu/drm/i915/gt/intel_engine_user.h
index f845ea1cbfaa..fc63d3994465 100644
--- a/drivers/gpu/drm/i915/gt/intel_engine_user.h
+++ b/drivers/gpu/drm/i915/gt/intel_engine_user.h
@@ -18,7 +18,7 @@ intel_engine_lookup_user(struct drm_i915_private *i915, u8 class, u8 instance);
 unsigned int intel_engines_has_context_isolation(struct drm_i915_private *i915);
 
 void intel_engine_add_user(struct intel_engine_cs *engine);
-void intel_engines_driver_register(struct drm_i915_private *i915);
+void __nocfi intel_engines_driver_register(struct drm_i915_private *i915);
 
 const char *intel_engine_class_repr(u8 class);
 
diff --git a/include/drm/drm_modes.h b/include/drm/drm_modes.h
index 99134d4f35eb..282f967fe727 100644
--- a/include/drm/drm_modes.h
+++ b/include/drm/drm_modes.h
@@ -524,7 +524,7 @@ drm_mode_validate_ycbcr420(const struct drm_display_mode *mode,
 			   struct drm_connector *connector);
 void drm_mode_prune_invalid(struct drm_device *dev,
 			    struct list_head *mode_list, bool verbose);
-void drm_mode_sort(struct list_head *mode_list);
+void __nocfi drm_mode_sort(struct list_head *mode_list);
 void drm_connector_list_update(struct drm_connector *connector);
 
 /* parsing cmdline modes */
-- 
2.17.1

